### PR TITLE
Benchmarks: Doc Revision - Add ReduceType into benchmarks doc

### PR DESCRIPTION
**Pull request by guoshzhao**
**Description**
Add ReduceType description into benchmarks doc.

### CODE REVIEW

File: docs/design-docs/benchmarks.md [hunk 1]
```diff
@@ -244,6 +249,12 @@ result = {
             'throughput-inference-float32': [avg_throughput1, ..., avg_throughputN],
             'throughput-inference-float16': [avg_throughput1, ..., avg_throughputN],
     },
```
Comment (cp5555) [inline docs/design-docs/benchmarks.md:251]: Do we need to change result as well?

File: docs/design-docs/benchmarks.md [hunk 2]
```diff
@@ -244,6 +249,12 @@ result = {
...
+    'reduce': {
```
Comment (cp5555) [inline docs/design-docs/benchmarks.md:252]: I don't find the code for the name. Can we change to `reduce_operator`? `reduce` is a little bit strange.

File: docs/design-docs/benchmarks.md [hunk 3]
```diff
@@ -181,7 +181,7 @@ This chapter will describe the interfaces with the caller (Superbench executor),
 
 The inputs needed by the `benchmarks` package is simple, just the context object of the benchmark want to run:
 
-### Invoke
+#### Invoke
```
Comment (cp5555) [inline docs/design-docs/benchmarks.md:184]: Why change this?

**Review by cp5555 — CHANGES_REQUESTED** [review]

File: docs/design-docs/benchmarks.md (hunk 2 above)
Comment (guoshzhao) [reply docs/design-docs/benchmarks.md:252]: It is one member of BenchmarkResult, and has been merged in last PR:
https://github.com/microsoft/superbenchmark/blob/main/superbench/benchmarks/result.py#L34

I will submit separated PR to modify it uniformly.

File: docs/design-docs/benchmarks.md (hunk 3 above)
Comment (guoshzhao) [reply docs/design-docs/benchmarks.md:184]: It should be `### Inputs`'s `#### Invoke`, just align with `### Outputs`

File: docs/design-docs/benchmarks.md (hunk 1 above)
Comment (guoshzhao) [reply docs/design-docs/benchmarks.md:251]: Here the result is from one executor, it don't need to change. `avg `here means the average of every step.

File: docs/design-docs/benchmarks.md (hunk 3 above)
Comment (cp5555) [reply docs/design-docs/benchmarks.md:184]: Got it!

File: docs/design-docs/benchmarks.md (hunk 1 above)
Comment (cp5555) [reply docs/design-docs/benchmarks.md:251]: I think the result should be min already. We don't have average anymore.

Comment (cp5555) [reply docs/design-docs/benchmarks.md:251]: Got it. It's for executor output.

**Review by cp5555 — APPROVED** [review]

**Review by TobeyQin — APPROVED** [review]

**Pull request merged**